packages/console/src/components/ActionMenu/use-position.ts

```diff
@@ -0,0 +1,43 @@
+import { RefObject, useCallback, useEffect, useLayoutEffect, useState } from 'react';
+
+type Position = {
+  left: number;
+  top: number;
+};
+
+const safePadding = 20;
+const overlayPadding = 4;
+
+export default function usePosition(
+  anchorReference: RefObject<HTMLElement>,
+  overlayReference: RefObject<HTMLElement>
+) {
+  const [position, setPosition] = useState<Position>({ left: 0, top: 0 });
+
+  const updatePosition = useCallback(() => {
+    if (anchorReference.current && overlayReference.current) {
+      const anchor = anchorReference.current.getBoundingClientRect();
+      const overlay = overlayReference.current.getBoundingClientRect();
+      const isTopSide =
+        anchor.y + anchor.height + overlay.height > window.innerHeight - safePadding;
+      const isLeftSide = anchor.x + overlay.width > window.innerWidth - safePadding;
+      const left = isLeftSide ? anchor.x + anchor.width - overlay.width : anchor.x;
+      const top = isTopSide
+        ? anchor.y - overlay.height - overlayPadding
+        : anchor.y + anchor.height + overlayPadding;
+      setPosition({ left, top });
+    }
+  }, [anchorReference, overlayReference]);
+
+  useLayoutEffect(updatePosition, [updatePosition]);
+
+  useEffect(() => {
+    window.addEventListener('resize', updatePosition);
```

Comment:
what about page scroll?

Reply:
we use "absolute" position (not "fixed") here, so page scroll is just good?